numpy/core/src/umath/loops.c.src

```diff
@@ -2618,32 +2618,33 @@ OBJECT_@kind@(char **args, npy_intp *dimensions, npy_intp *steps, void *NPY_UNUS
 NPY_NO_EXPORT void
 OBJECT_sign(char **args, npy_intp *dimensions, npy_intp *steps, void *NPY_UNUSED(func))
 {
-#if defined(NPY_PY3K)
     PyObject *zero = PyLong_FromLong(0);
+    PyObject *one = PyLong_FromLong(1);
+    PyObject *minus_one = PyLong_FromLong(-1);
+
     UNARY_LOOP {
         PyObject *in1 = *(PyObject **)ip1;
         PyObject **out = (PyObject **)op1;
-        int v;
+        int is_eq, is_gt;
         PyObject *ret;
-        if (PyObject_Cmp(in1 ? in1 : Py_None, zero, &v) == -1) {
+
+        is_gt = PyObject_RichCompareBool(in1 ? in1 : Py_None, zero, Py_GT);
+        is_eq = PyObject_RichCompareBool(in1 ? in1 : Py_None, zero, Py_EQ);
```

Comment:
I am doing a second comp because, if the no. is not greater than zero, how would we know if its equal to or less than zero ?